qemu/tests/fio_windows.py

```diff
@@ -33,31 +33,43 @@ def run(test, params, env):
 
     vm = env.get_vm(params["main_vm"])
     vm.verify_alive()
-    session = vm.wait_for_login(timeout=timeout)
+    session_sys = vm.wait_for_login(timeout=timeout)
+    session_data = vm.wait_for_login(timeout=timeout)
+
+    error_context.context("Format disk", logging.info)
+    utils_misc.format_windows_disk(session_sys, params["disk_index"],
+                                   mountpoint=params["disk_letter"])
     try:
-        installed = session.cmd_status(check_installed_cmd) == 0
+        installed = session_sys.cmd_status(check_installed_cmd) == 0
         if not installed:
-            dst = r"%s:\\" % utils_misc.get_winutils_vol(session)
+            dst = r"%s:\\" % utils_misc.get_winutils_vol(session_sys)
 
             error_context.context("Install fio in guest", logging.info)
             install_cmd = params["install_cmd"]
             install_cmd = re.sub(r"DRIVE:\\+", dst, install_cmd)
-            session.cmd(install_cmd, timeout=180)
+            session_sys.cmd(install_cmd, timeout=180)
             time.sleep(30)
             config_cmd = params.get("config_cmd")
             if config_cmd:
-                session.cmd(config_cmd)
+                session_sys.cmd(config_cmd)
 
         error_context.context("Start fio in guest.", logging.info)
-        s, o = session.cmd_status_output(fio_cmd, timeout=(cmd_timeout+60))
-        if s:
-            raise exceptions.TestError("Failed to run fio, output: %s" % o)
+        fio_thread_data = utils_misc.InterruptedThread(session_data.cmd_status_output,
+                                                       (fio_cmd_data, (cmd_timeout+600)))
```

Comment:
I prefer keeping the original expression. You can enlarge the timeout by editing `cmd_timeout`.

Reply:
Updated, thanks~